### ISSUE TITLE
SharedWithMe is not supported by SDK

### ISSUE THREAD

**Issue by thechandanbhagat** [OneDrive/onedrive-sdk-csharp#133]
I have been trying OneDriveClient.Drive.Shared.Request().GetAsync() to get the list. But this doesnot return data of my shared folder. Is there something I am missing?

**Comment by speedchandu** [comment]
@cdmayer Actually, i get the response! but i am unable to figure out the how do we enlist the shared documents/folders to us. Can you guide me through it? 
I have done as following
```
var fold = await this.oneDriveClient.Drive.Shared.Request().GetAsync();
var list = fold.CurrentPage.Select(p => new DriveData
                            {
                                SelfID = p.Id,
                                Name = p.Name,
                                IsFolder = p.Folder != null ? true : false,
                                IsFile = p.File != null ? true : false,
                                MimeType = p.File != null ? p.File.MimeType : "Folder"
                            }).ToList();
```
but doing this just gives me the ID and Name and nothing else. Am i missing something here?

**Comment by cdmayer** [comment]
Not sure what you are trying to accomplish, but the code you have is doing what it's supposed to do :) Check out ["List items shared from a drive"](https://dev.onedrive.com/drives/shared_by_me.htm) - basically you're getting a list of the files you have shared. If you want more information about a given file, you would do:

```
var item = await oneDriveClient
                     .Items[list[0].SelfID]
                     .Request()
                     .GetAsync();
```

By the way, why are you transforming `Item`s into `DriveData`? Seems like you are losing information in the process (because an `Item` can have a lot of information on it other than your `DriveData` type).

**Comment by speedchandu** [comment]
@cdmayer Actually, if that is the case that i am getting the list of files I have shared. How do i get the list of files/folders shared to me. I have nothing to do with the files i have shared. All I need is to get the information about the items shared to me. Can you suggest me how do i achieve this?
`DriveData` is the model that i am populating from SDK to mine. Its because to populate in my application. I am not getting any idea how do i get the list of shared files/folders to me.

**Comment by cdmayer** [comment]
Unfortunately you have identified a bug. You should be able to access things "[shared with me](https://dev.onedrive.com/drives/shared_with_me.htm)", but that is not available right now in the SDK. I'll tag this as a bug to be fixed.

**Comment by cdmayer** [comment]
Until this bug is fixed, you can build a new request by subclassing `BaseRequest` and `BaseRequestBuilder`. First you would build a request using `client.Drive`, then append a new segment with `view.sharedWithMe`. Take a look at how the other `RequestBuilder` classes do it to get an idea.

**Comment by jclarkswiftsoftware** [comment]
I'm also running into this issue.  Are there plans to update the SDK with the missing functionality in the next few months?

**Comment by jclarkswiftsoftware** [comment]
If I update the SDK with a fix for this issue, will Microsoft accept a pull request for the changes?

**Comment by cdmayer** [comment]
Submit it! We can review the changes together, too.

**Comment by jclarkswiftsoftware** [comment]
Ok thanks.  I've successfully been able to update the SDK with the missing functionality.  I had trouble integrating with the authentication changes in version 2.0.0 of the SDK, so I integrated my code changes with version 1.2.0.  

(The OneDrive Webhooks example code, which I based my code on, does not integrate with version 2.0.0 of the SDK).  

Since this is the case, would you still like me to do a pull request for my changes?

**Comment by cdmayer** [comment]
do a pull request into https://github.com/OneDrive/onedrive-sdk-csharp/tree/v1-maintenance

**Comment by brucewangkorea** [comment]
Anyone succeeded?

In my case it just throw Exceptions, contrary to @speedchandu.

this.oneDriveClient.Shares.Request().GetAsync()
--> {Code: BadRequest
Message: Unsupported segment type. ODataQuery: users/aaa@aaa.com/shares
} Microsoft.Graph.Error

this.oneDriveClient.Drive.Shared.Request().GetAsync()
--> {Code: BadRequest
Message: Resource not found for the segment 'shared'.
} Microsoft.Graph.Error

**Comment by MortenChristiansen** [comment]
The following works for me:
```csharp
private class Data
{
    public Item[] Value { get; set; }
}

public static async Task<List<Item>> GetFilesInSharedWithMe(this IOneDriveClient service)
{
    var request = new HttpRequestMessage(HttpMethod.Get, $"{service.BaseUrl}/drive/view.sharedWithMe");
    await service.AuthenticationProvider.AppendAuthHeaderAsync(request).ConfigureAwait(false);
    var response = service.HttpProvider.SendAsync(request).Result;
    var deserialized = service.HttpProvider.Serializer.DeserializeObject<Data>(await response.Content.ReadAsStreamAsync().ConfigureAwait(false));
    return deserialized.Value.Select(i => i.RemoteItem).ToList();
}
```

though I could not get the `request.Select(Fields)` to work properly.

**Comment by speedchandu** [comment]
@MortenChristiansen this is working.

**Comment by speedchandu** [comment]
@brucewangkorea Please refer to solution that is below. This solution really helped me.

**Comment by giobiondani** [comment]
I installed the solution kindly provided by @jclarkswiftsoftware but I was getting an error with the SharedWithMe request: "invalid request - API not found".
In DriveRequestBuilder.cs, I replaced this:
`return new SharedWithMeCollectionRequestBuilder(this.AppendSegmentToRequestUrl("sharedWithMe"), this.Client); `
with this:
`return new SharedWithMeCollectionRequestBuilder(this.AppendSegmentToRequestUrl("view.sharedWithMe"), this.Client); 
` 
now it works and brings up the list of items shared with me.
However, not all items are returned - when I look on the website I can see 10 items shared with me, but the API brings up only 8. I can't see anything special about the 2 that are missing.
Are you aware of any kind of filtering that is done by view.sharedWithMe?
